hbase/src/main/java/org/apache/metamodel/hbase/HBaseUpdateCallback.java

```diff
@@ -0,0 +1,137 @@
+    }
+
+    public HBaseCreateTableBuilder createTable(Schema schema, String name, HBaseColumn[] outputColumns)
+            throws IllegalArgumentException, IllegalStateException {
+        return new HBaseCreateTableBuilder(this, schema, name, outputColumns);
+    }
+
+    @Override
+    public boolean isDropTableSupported() {
+        return true;
+    }
+
+    @Override
+    public TableDropBuilder dropTable(Table table) throws IllegalArgumentException, IllegalStateException,
+            UnsupportedOperationException {
+        return new HBaseTableDropBuilder(table, this);
+    }
+
+    public void dropTableExecute(Table table) {
+        try {
+            final HBaseWriter HbaseWriter = new HBaseWriter(HBaseDataContext.createConfig(_configuration));
+            HbaseWriter.dropTable(table.getName());
+            MutableSchema schema = (MutableSchema) table.getSchema();
+            schema.removeTable(table);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public RowInsertionBuilder insertInto(Table table) throws IllegalArgumentException, IllegalStateException,
+            UnsupportedOperationException {
+        throw new UnsupportedOperationException("Use insertInto(Table table, HBaseColumn[] outputColumns)");
+    }
+
+    public HBaseRowInsertionBuilder insertInto(Table table, HBaseColumn[] columns) throws IllegalArgumentException {
+        if (table instanceof HBaseTable) {
+            return new HBaseRowInsertionBuilder(this, (HBaseTable) table, columns);
+        }
+        throw new IllegalArgumentException("Not an HBase table: " + table);
+    }
+
+    protected synchronized void writeRow(HBaseTable hBaseTable, HBaseColumn[] outputColumns, Object[] values) {
+        try {
+            final HBaseWriter HbaseWriter = new HBaseWriter(HBaseDataContext.createConfig(_configuration));
+            HbaseWriter.writeRow(hBaseTable, outputColumns, values);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public boolean isDeleteSupported() {
+        return false;
+    }
+
+    @Override
+    public RowDeletionBuilder deleteFrom(Table table) throws IllegalArgumentException, IllegalStateException,
+            UnsupportedOperationException {
+        throw new UnsupportedOperationException(
+                "Use deleteFrom(HBaseUpdateCallback updateCallback, Table table, Object key)");
+    }
+
+    public HBaseRowDeletionBuilder deleteFrom(HBaseUpdateCallback updateCallback, Table table)
+            throws IllegalArgumentException {
+        if (table instanceof HBaseTable) {
+            return new HBaseRowDeletionBuilder(this, (HBaseTable) table);
+        }
+        throw new IllegalArgumentException("Not an HBase table: " + table);
+    }
+
+    protected synchronized void deleteRow(HBaseTable hBaseTable, Object key) {
```

Comment:
Same here as for the `writeRow` method, why is this method `protected`? To me `protected` indicates that it is a method which is typically overridden in a sub-class, but that's not the case here, as far as I can tell you only need it to be accessible from the `HBaseRowDeletionBuilder`. So than it could also be package protected instead, but I personally think it would even be better to remove this method and move its logic into the `HBaseRowDeletionBuilder#execute` method.